### PR TITLE
Root review

### PR DESCRIPTION
**Pull request by bencbernstein**
- Feels good to me, definitely needs some testing
- TODO: Don't show a hex in the hive if `root.words < 3`

### CODE REVIEW

**Comment by bencbernstein** [comment]
can you take a look at that `GameManager` conflict?

**Comment by bencbernstein** [comment]
```mutating func nextQuestion(not: Word) -> Question? {
    questionCount += 1

    var word: Word!

    if let user = Datastore.shared.user, !user.completedTutorial, let tutorialWord = user.tutorialWords.first {
      if let nextWord = Datastore.shared.wordForString(tutorialWord.value) { word = nextWord }
      RealmManager.deleteTutorialWord()
    } else {
      var words: [Word]
      isRootReview ? (words = Datastore.shared.rootReviewWords) : (words = Datastore.shared.words)
      if let randomWord = words.filter({ $0 != not }).randomElement() {
        word = randomWord
      } else {
        return nil
      }
    }
    
    if Datastore.shared.isNewWord(word: word) { newWordsLearned += 1 }
    return Question(word: word, buttonMode: buttonMode)
  }
``` i think that resolves it but i was getting a few errors, might have been a realm thing

File: Apiary/Managers/GameManager.swift [hunk 1]
```diff
@@ -126,16 +127,18 @@ struct GameManager {
     return completion.value > 1
   }
   
-  init(user: User?, isSpeedRound: Bool, minQtoComplete: Int = 10) {
+  init(user: User?, isSpeedRound: Bool, minQtoComplete: Int = 10, isRootReview: Bool) {
```
Comment (OLI9292) [inline Apiary/Managers/GameManager.swift]: can this be refactored to us the RoundType enum instead?

File: Apiary/Managers/GameManager.swift [hunk 2]
```diff
@@ -153,15 +156,18 @@ extension GameManager {
...
+
     var word: Word!
 
     if let user = Datastore.shared.user, !user.completedTutorial, let tutorialWord = user.tutorialWords.first {
       if let nextWord = Datastore.shared.wordForString(tutorialWord.value) { word = nextWord }
       RealmManager.deleteTutorialWord()
     } else {
-      if let randomWord = Datastore.shared.words.filter({ $0 != not }).randomElement() {
+      var words: [Word]
+      isRootReview ? (words = Datastore.shared.rootReviewWords) : (words = Datastore.shared.words)
```
Comment (OLI9292) [inline Apiary/Managers/GameManager.swift]: might make more sense to hold the rootReviewWords in the struct, the Datastore is more for static data

File: Apiary/Managers/GameManager.swift [hunk 3]
```diff
@@ -172,8 +178,16 @@ extension GameManager {
     return Question(word: word, buttonMode: buttonMode)
   }
   
-  mutating func reset(type: RoundType) {
+    if Datastore.shared.isNewWord(word: word) { newWordsLearned += 1 }
+    return Question(word: word, buttonMode: buttonMode)
+  }
+  
+  mutating func set(type: RoundType) {
```
Comment (OLI9292) [inline Apiary/Managers/GameManager.swift]: could have a better name but i do prefer reset over set as it's a lil more explicit that its resetting for the next round

File: Apiary/ViewControllers/HiveViewController.swift [hunk 4]
```diff
@@ -31,7 +31,8 @@ final class HiveViewController: UIViewController {
       layout()
     } else {
       startSpinner()
-      store.fetchWords() { _ in self.layout() }
+      store.fetchWords() { _ in self.layout()
+      }
```
Comment (OLI9292) [inline Apiary/ViewControllers/HiveViewController.swift]: this `}` got indented

File: Apiary/Managers/GameManager.swift (hunk 1 above)
Comment (bencbernstein) [inline Apiary/Managers/GameManager.swift]: so the game manager holds a round type on it?

File: Apiary/ViewControllers/HiveViewController.swift [hunk 5]
```diff
@@ -185,6 +191,17 @@ final class HiveViewController: UIViewController {
     }
   }
   
+  @objc private func rootHexTapped(_ gestureRecognizer: UITapGestureRecognizer) {
```
Comment (OLI9292) [inline Apiary/ViewControllers/HiveViewController.swift]: hex.subviews[0] is a force unwrap

i find chaining the guard lets reads more simply, like
```guard 
  let hex = gestureRecognizer.view as? Hexagon,
  let hex.first() as? UILabel,
  let rootStr = rootLabel.text
  else { return}```

File: Apiary/Managers/GameManager.swift (hunk 1 above)
Comment (OLI9292) [inline Apiary/Managers/GameManager.swift]: ya, then you can get rid of `isSpeedRound` and `isRootReview`

**Pull request merged**